packages/calypso-e2e/src/lib/pages/themes-page.ts

```diff
@@ -58,28 +58,42 @@ export class ThemesPage extends BaseContainer {
 	 */
 	async search( keyword: string ): Promise< void > {
 		const searchInput = await this.page.waitForSelector( selectors.searchInput );
-		await searchInput.fill( keyword );
+		await Promise.all( [ this.page.waitForNavigation(), searchInput.fill( keyword ) ] );
 		await this.page.waitForSelector( selectors.placeholder, { state: 'hidden' } );
 	}
 
 	/**
-	 * Select a theme by name.
+	 * Selects a theme from the gallery.
 	 *
-	 * If the theme is not shown, this will throw an Error.
+	 * This method can perform either exact name match or a fuzzy (partial) name match.
+	 * To match exactly, supply the full name of the theme as it appears to the user.
+	 * For a fuzzy match, supply partially the name of the them that should be matched.
 	 *
-	 * @param {string} name Theme name to select.
+	 * Example:
+	 * 		partial match: `Twenty Twen` -> [Twenty Twenty, Twenty Twenty-One]
+	 * 		exact match: `Twenty Seventeen` -> Twenty Seventeen
+	 *
+	 * @param {string} [name] Theme name to select.
 	 * @returns {Promise<void>} No return value.
-	 * @throws {Error} If theme is not shown on page.
 	 */
 	async select( name: string ): Promise< void > {
+		// Build selector that will select themes on the page that match the name but excludes
+		// currently activated themes.
+		const selector = `${ selectors.items }:has-text("${ name }")${ selectors.excludeActiveTheme }`;
+		// Get number of themes being shown on page.
+		const numThemes = await this.page.$$( selector ).then( ( themes ) => themes.length );
```

Comment:
I might be misunderstanding how the empty case is handled here, but do we want to through if numThemes is 0?

I worry that the behavior of trying to use the index of 0 later in the waitForSelector might result in a more confusing error message than one that explicitly says something like "No themes were found matching the search name".

Reply:
Added case to handle if `numTheme === 0`.